app/views/posts/new.html.erb

```diff
@@ -1,6 +1,6 @@
 <h2>Create new post</h2>
 
-<%= form_with model: [@user, @post] do |form| %>
+<%= form_with model: [@user, @post], url: user_posts_path(@user) do |form| %>
```

Comment:
- [ ] I get the below error when I try to create a post. Please fix the error so a user can add a post. Also, ensure that your set of rules for user-post interactions. are working i.e:

* A user can delete a post if it is theirs or if they have an admin role
* A user can delete a comment if it is theirs or if they have an admin role